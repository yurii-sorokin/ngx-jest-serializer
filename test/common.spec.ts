import { initHostFixture } from './framework/host'

describe(`Common`, () => {
  it(`should print list generated by ngFor`, () => {
    const fixture = initHostFixture({
      props: {
        trackById: (index, item) => item.id,
        items: [
          { id: 1, text: 'text1' },
          { id: 2, text: 'text2' },
          { id: 3, text: 'text3' },
          { id: 4, text: 'text4' },
        ]
      },
      template:
        `<div
          *ngFor="let item of items; trackBy: trackById"
        >
          <div>{{item.text}}</div>
        </div>`
    })

    expect(fixture).toMatchSnapshot()
  })

  it('should print embedded views', () => {
    const fixture = initHostFixture({
      props: {
        context: { $implicit: 'World', localSk: 'Svet' }
      },
      template:
        `<ng-container *ngTemplateOutlet="hello"></ng-container>
        <ng-container *ngTemplateOutlet="helloEng; context: context"></ng-container>
        <ng-container *ngTemplateOutlet="helloSvk; context: context"></ng-container>
        <ng-template #hello><span>Hello!</span></ng-template>
        <ng-template #helloEng let-name><span>Hello {{name}}!</span></ng-template>
        <ng-template #helloSvk let-world="localSk"><span>Ahoj {{world}}!</span></ng-template>`
    })

    expect(fixture).toMatchSnapshot()
  })
})
