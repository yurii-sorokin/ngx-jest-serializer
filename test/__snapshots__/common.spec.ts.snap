// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common should print embedded views 1`] = `
<tst-host>
  <span>
    Hello!
  </span>
  
        
  <span>
    Hello World!
  </span>
  
        
  <span>
    Ahoj Svet!
  </span>
  
        
  
        
  
        
</tst-host>
`;

exports[`Common should print list generated by ngFor 1`] = `
<tst-host>
  <div>
    
          
    <div
      class="content"
    >
      text1
    </div>
    
        
  </div>
  <div>
    
          
    <div
      class="content"
    >
      text2
    </div>
    
        
  </div>
  <div>
    
          
    <div
      class="content"
    >
      text3
    </div>
    
        
  </div>
  <div>
    
          
    <div
      class="content"
    >
      text4
    </div>
    
        
  </div>
</tst-host>
`;
